builtin/providers/vault/resource_mount.go

```diff
@@ -0,0 +1,133 @@
+package vault
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/vault/api"
+	"log"
+)
+
+func mountResource() *schema.Resource {
+	return &schema.Resource{
+		Create: mountWrite,
+		Update: mountUpdate,
+		Delete: mountDelete,
+		Read:   mountRead,
+
+		Schema: map[string]*schema.Schema{
+			"path": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Where the secret backend will be mounted",
+			},
+
+			"type": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Type of the backend, such as 'aws'",
+			},
+
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Required:    false,
+				ForceNew:    true,
+				Description: "Human-friendly description of the mount",
+			},
+
+			"default_lease_ttl_seconds": {
+				Type:        schema.TypeInt,
+				Required:    false,
+				Optional:    true,
+				ForceNew:    false,
+				Description: "Default lease duration for tokens and secrets in seconds",
+			},
+
+			"max_lease_ttl_seconds": {
+				Type:        schema.TypeInt,
+				Required:    false,
+				Optional:    true,
+				ForceNew:    false,
+				Description: "Maximum possible lease duration for tokens and secrets in seconds",
+			},
+		},
+	}
+}
+
+func mountWrite(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	info := &api.MountInput{
+		Type:        d.Get("type").(string),
+		Description: d.Get("description").(string),
+		Config: api.MountConfigInput{
+			DefaultLeaseTTL: fmt.Sprintf("%ds", d.Get("default_lease_ttl_seconds")),
+			MaxLeaseTTL:     fmt.Sprintf("%ds", d.Get("max_lease_ttl_seconds")),
+		},
+	}
+
+	path := d.Get("path").(string)
+
+	log.Printf("[DEBUG] Writing mount %s to Vault", path)
+
+	if err := client.Sys().Mount(path, info); err != nil {
+		return fmt.Errorf("error writing to Vault: %s", err)
+	}
+
+	d.SetId(path)
+
+	return nil
+}
+
+func mountUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	config := api.MountConfigInput{
+		DefaultLeaseTTL: fmt.Sprintf("%ds", d.Get("default_lease_ttl_seconds")),
+		MaxLeaseTTL:     fmt.Sprintf("%ds", d.Get("max_lease_ttl_seconds")),
+	}
+
+	path := d.Id()
+
+	log.Printf("[DEBUG] Updating mount %s in Vault", path)
+
```

Comment:
It _is_ actually valid to call `d.SetId` from inside the update function, so in principle it should be fine to support remount here and just be careful do call `d.SetId` if and only if the remount succeeds.

This seems desirable to allow moving a mount without losing its internal state, though if you'd rather save that for a followup PR I think that's fine.